pkg/wal/processor/transformer/wal_transformer.go

```diff
@@ -0,0 +1,137 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package transformer
+
+import (
+	"context"
+	"fmt"
+
+	loglib "github.com/xataio/pgstream/pkg/log"
+	"github.com/xataio/pgstream/pkg/transformers"
+	"github.com/xataio/pgstream/pkg/transformers/builder"
+	"github.com/xataio/pgstream/pkg/wal"
+	"github.com/xataio/pgstream/pkg/wal/processor"
+)
+
+// Transformer is a decorator around a wal processor that transforms wal event
+// column values following the configured transformation rules.
+type Transformer struct {
+	logger         loglib.Logger
+	processor      processor.Processor
+	transformerMap map[string]columnTransformers
+}
+
+type columnTransformers map[string]transformers.Transformer
+
+type Config struct {
+	TransformerRulesFile string
+}
+
+type Option func(t *Transformer)
+
+// New will return a transformer processor wrapper that will transform incoming
+// wal event column values as configured by the transformation rules.
+func New(cfg *Config, processor processor.Processor, opts ...Option) (*Transformer, error) {
+	rules, err := readRulesFromFile(cfg.TransformerRulesFile)
+	if err != nil {
+		return nil, err
```

Comment:
maybe we want to add some context to the error here

Reply:
The `readRulesFromFile` function returns already context within the errors:
https://github.com/xataio/pgstream/blob/add-transformer-processor/pkg/wal/processor/transformer/wal_transformer_rules.go#L30-L42